apps/api/src/app/integrations/usecases/remove-integration/remove-integration.usecase.ts

```diff
@@ -25,6 +25,10 @@ export class RemoveIntegration {
         throw new NotFoundException(`Entity with id ${command.integrationId} not found`);
       }
 
+      if (this.isNovuIntegration(existingIntegration)) {
+        throw new ApiException('Novu demo integration or in-app integration cannot be deleted');
```

Comment:
Prefer a 4XX error rathen than a 5xx for this scenario as this is an expected business logic.
```suggestion
        throw new BadRequest('Novu demo integration or in-app integration cannot be deleted');
```